pkg/components/exporter/exporter.go

```diff
@@ -509,3 +511,93 @@ func NewServiceMonitor() *monv1.ServiceMonitor {
 		},
 	}
 }
+
+func NewPrometheusRule() *monv1.PrometheusRule {
+	interval := monv1.Duration("15s")
+
+	return &monv1.PrometheusRule{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: monv1.SchemeGroupVersion.String(),
+			Kind:       "PrometheusRule",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PrometheusRuleName,
+			Namespace: components.Namespace,
+			Labels:    labels,
+		},
+		Spec: monv1.PrometheusRuleSpec{
+			Groups: []monv1.RuleGroup{{
+				Name:     "kepler.rules",
+				Interval: &interval,
+				Rules: []monv1.Rule{
+					record("kepler:container_joules_total:consumed:24h:all",
+						`sum(
+							increase(kepler_container_joules_total[24h:1m])
+						)`,
+					),
+					record("kepler:container_joules_total:consumed:24h:by_ns",
+						`sum by (container_namespace) (
+								increase(kepler_container_joules_total[24h:1m])
+						 )`,
+					),
+
+					record("kepler:container_gpu_joules_total:consumed:1h:by_ns",
+						`sum by (container_namespace) (
+								increase(kepler_container_gpu_joules_total[1h:15s])
+						 )`,
+					),
+
+					record("kepler:container_dram_joules_total:consumed:1h:by_ns",
+						`sum by (container_namespace) (
+								increase(kepler_container_dram_joules_total[1h:15s])
+						 )`,
+					),
+
+					record("kepler:container_package_joules_total:consumed:1h:by_ns",
+						`sum by (container_namespace) (
+								increase(kepler_container_package_joules_total[1h:15s])
+						 )`,
+					),
+
+					record("kepler:container_other_joules_total:consumed:1h:by_ns",
+						`sum by (container_namespace) (
+								increase(kepler_container_other_joules_total[1h:15s])
+						 )`,
+					),
+
+					// irate of joules = joules per second -> watts
+					record("kepler:container_gpu_watts:1m:by_ns_pod",
+						`sum by (container_namespace, pod_name) (
+								irate(kepler_container_gpu_joules_total[1m])
+						)`,
+					),
+
+					record("kepler:container_package_watts:1m:by_ns_pod",
+						`sum by (container_namespace, pod_name) (
+								irate(kepler_container_package_joules_total[1m])
+						)`,
+					),
+
+					record("kepler:container_other_watts:1m:by_ns_pod",
+						`sum by (container_namespace, pod_name) (
+								irate(kepler_container_other_joules_total[1m])
+						)`,
+					),
+
+					record("kepler:container_dram_watts:1m:by_ns_pod",
+						`sum by (container_namespace, pod_name) (
```

Comment:
Just a small question is there any reason to why we are doing sum by `container_namespace`? I see in the queries we are only doing sum by `pod_name`.

Reply:
The query also needs `container_namespace`